dxr/static/js/code-highlighter.js

```diff
@@ -115,21 +115,25 @@ $(function () {
         }
     }
 
-    //update places where hash location is used: window, permalink
+    //update places where hash location is used: window, sync-links
     function updateHash(hash) {
-        if (permalink.length > 0)
-            updatePermalink(hash);
+        syncLinks.each(function() {
+            updateLink($(this), hash);
+        });
         history.replaceState(null, '', hash);
     }
 
-    //update the permalink href based on the windowHash.
-    function updatePermalink(windowHash) {
-        var permalink_href = permalink.attr('href'),
-            hash_loc = permalink_href.indexOf('#');
-        // If the link already has #, then cut that it off.
+    //update the link href based on the windowHash.
+    function updateLink(anchor, windowHash) {
+        var href = anchor.attr('href'),
+            hash_loc = href.indexOf('#');
+        // If the link already has #, then cut it off.
         if (hash_loc >= 0)
-            permalink_href = permalink_href.substring(0, hash_loc);
-        permalink.attr('href', permalink_href + windowHash);
+            href = href.substring(0, hash_loc);
+        // hg web uses #l and GitHub uses #L but supports #l, so we use #l here.
```

Comment:
I was thinking we'd stop short of supporting multiple ranges, since nothing else can make use of them atm. That's how I still lean.

But you got me thinking again about the problem of constructing multi-region ranges for outside systems, when we can't predict how they'd format them. After a brief detour through simple sets of tokens we could provide to do fake boolean logic, it occurred to me that we already have a safe, client-side language available for building strings that can do conditionals and everything: nunjucks. So, if we do any sort of text substitution at all, I think we should spell the tokens in a Jinja-compatible way: `{% smoo %}`. Then at least we'll be forward compatible if we do support multiple ranges someday. (Can nunjucks parse a template client-side? I think it can.)